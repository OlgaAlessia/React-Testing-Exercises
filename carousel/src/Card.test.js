import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

// smoke test
test("renders without crashing", function () {
  render(<Card caption="Test Photo" src="testImage.png" currNum={1} totalNum={1} />);
});

// snapshot test
test("matches snapshot", function () {
  const { asFragment } = render(<Card caption="Test Photo" src="testImage.png" currNum={1} totalNum={1} />);
  expect(asFragment()).toMatchSnapshot();
});
