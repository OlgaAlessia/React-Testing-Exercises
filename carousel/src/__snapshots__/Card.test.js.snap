// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Test Photo
    </h4>
    <img
      alt="Test Photo"
      class="Card-image"
      src="testImage.png"
    />
    <small
      class="Card-small"
    >
      Image 1 of 1.
    </small>
  </div>
</DocumentFragment>
`;
